readiness-tool/popup.js

```diff
@@ -0,0 +1,273 @@
+ * @return {Object} 
+ */
+function findDetectedApps(html) {
+  fetch('apps.json')
+  .then(response => {
+    return response.json().then(data => {
+      if (response.ok) {
+        let listAllApps = data.apps;
+        detectedApps = filterApps(html, listAllApps);
+        showSupportedAppsInView(detectedApps, listAllApps);
+        return detectedApps;
+      } else {
+        return Promise.reject({status: response.status});
+      }
+    });
+  })
+  .catch(error => console.error('apps.json in the readiness tool is invalid.', 
+                                error));
+  
+}
+
+/**
+ * Add supported and unsupported applications to the view
+ * @param {Object} detectedApps - All 3rd Party Applications found on page
+ * @param {!Object} listAllApps - JSON of all 3p vendors
+ */
+function showSupportedAppsInView(detectedApps, listAllApps) {
+  supportedAds.textContent = supportedAnalytics.textContent =
+    notSupportedAds.textContent = notSupportedAnalytics.textContent =
+    blankMessage;
+  supportedAds.appendChild(makeList(detectedApps.supported.ads, false,
+    listAllApps));
+  supportedAnalytics.appendChild(makeList(detectedApps.supported.analytics,
+    false, listAllApps));
+  notSupportedAds.appendChild(makeList(detectedApps.notSupported.ads, true,
+    listAllApps));
+  notSupportedAnalytics.appendChild(makeList(
+    detectedApps.notSupported.analytics,
+    true, listAllApps));
+}
+
+
+/**
+ * Splits all detected apps into 'supported' and 'not supported'
+ * @param {string} htmlString - String containing all HTML on the page
+ * @param {!Object} listAllApps - JSON of all the 3p vendors 
+ * @return {Object} 
+ */
+function filterApps(htmlString, listAllApps) {
+  const filteredApps = {
+    'supported': {
+      'ads': [],
+      'analytics': []
+    },
+    'notSupported': {
+      'ads': [],
+      'analytics': []
+    }
+  };
+  // for all the app objects in the apps.JSON file
+  Object.keys(listAllApps).forEach(function (appName) {
+    const appConfig = listAllApps[appName];
+    // If object has a 'script' key
+    if (appConfig.script) {
+      appConfig.script.forEach(function (x) {
+        if (appConfig.cats.length == 0) {
+          console.error('The app', appName,
+            'does not have a value for "cats" in apps.json');
+          return;
+        } else if (appConfig.cats != "36" && appConfig.cats != "10") {
+          console.error('The app',
+            appName,
+            'is not declared as an ads or analytics vendor in apps.json');
+          return;
+        }
+        addToDict(x.split(appsRegexDelimeter)[0], htmlString, filteredApps, appName,
+          appConfig.cats[0]);
+      });
+    }
+  });
+  return filteredApps;
+}
+
+/**
+ * An object contained by 2 more objects that each hold two string arrays
+ * @typedef {{[String][String]}{[String][String]}} filteringObject
+ */
+
+/**
+ * Pushes app names to the supported or not supported list of the object
+ 'found this'
```

Comment:
This is outdated now that we've renamed the variable to `filteredApps`. Also, comments should start with a `*` on the new line.